crates/sui-core/src/authority/authority_per_epoch_store.rs

```diff
@@ -4530,6 +4530,33 @@ impl AuthorityPerEpochStore {
             .map(|s| s.summary))
     }
 
+    pub(crate) fn get_lowest_non_genesis_checkpoint_summary(
+        &self,
+    ) -> SuiResult<Option<CheckpointSummary>> {
+        let tables = self.tables()?;
+
+        // Get all sequence numbers, sort them, and find the lowest non-genesis one
+        let mut sequence_numbers: Vec<_> = tables
+            .builder_checkpoint_summary_v2
+            .safe_iter_with_bounds(None, None)
+            .map(|result| result.map(|(seq, _)| seq))
+            .collect::<Result<Vec<_>, _>>()?;
```

Comment:
We don't want to scan the whole table just to get the smallest element. `safe_iter_with_bounds(None, None).find(|seq| seq > 0)` should do the trick